dlt/destinations/synapse/synapse.py

```diff
@@ -0,0 +1,222 @@
+        "decimal": "decimal(%i,%i)",
+        "time": "time(%i)",
+        "wei": "decimal(%i,%i)"
+    }
+
+    dbt_to_sct = {
+        "nvarchar": "text",
+        "float": "double",
+        "bit": "bool",
+        "datetimeoffset": "timestamp",
+        "date": "date",
+        "bigint": "bigint",
+        "varbinary": "binary",
+        "decimal": "decimal",
+        "time": "time",
+        "tinyint": "bigint",
+        "smallint": "bigint",
+        "int": "bigint",
+    }
+
+HINT_TO_SYNAPSE_ATTR: Dict[TColumnHint, str] = {
+    "unique": "UNIQUE"
+}
+
+
+class SynapseStagingCopyJob(SqlStagingCopyJob):
+
+    @classmethod
+    def generate_sql(cls, table_chain: Sequence[TTableSchema], sql_client: SqlClientBase[Any]) -> List[str]:
+        sql: List[str] = []
+        for table in table_chain:
+            with sql_client.with_staging_dataset(staging=True):
+                staging_table_name = sql_client.make_qualified_table_name(table["name"])
+            table_name = sql_client.make_qualified_table_name(table["name"])
+            # drop destination table
+            sql.append(f"DROP TABLE {table_name};")
+            # moving staging table to destination schema
+            sql.append(f"ALTER SCHEMA {sql_client.fully_qualified_dataset_name()} TRANSFER {staging_table_name};")
+        return sql
+
+
+class SynapseMergeJob(SqlMergeJob):
+    @classmethod
+    def gen_key_table_clauses(cls, root_table_name: str, staging_root_table_name: str, key_clauses: Sequence[str], for_delete: bool) -> List[str]:
+        """Generate sql clauses that may be used to select or delete rows in root table of destination dataset
+        """
+        if for_delete:
+            # MS SQL doesn't support alias in DELETE FROM
+            return [f"FROM {root_table_name} WHERE EXISTS (SELECT 1 FROM {staging_root_table_name} WHERE {' OR '.join([c.format(d=root_table_name,s=staging_root_table_name) for c in key_clauses])})"]
+        return SqlMergeJob.gen_key_table_clauses(root_table_name, staging_root_table_name, key_clauses, for_delete)
+
+    @classmethod
+    def _to_temp_table(cls, select_sql: str, temp_table_name: str) -> str:
+        return f"SELECT * INTO {temp_table_name} FROM ({select_sql}) as t;"
+
+    @classmethod
+    def _new_temp_table_name(cls, name_prefix: str) -> str:
+        name = SqlMergeJob._new_temp_table_name(name_prefix)
+        return '#' + name
+
+class SynapseInsertValuesLoadJob(InsertValuesLoadJob):
+
+    def __init__(self, table_name: str, file_path: str, sql_client: SqlClientBase[Any]) -> None:
+        # First, set any attributes specific to this subclass
+        self._sql_client = sql_client
+        self._file_name = FileStorage.get_file_name_from_file_path(file_path)
+
+        # Then, call the parent class's __init__ method with the required arguments
+        super().__init__(table_name, file_path, sql_client)
+
+    def _insert(self, qualified_table_name: str, file_path: str) -> Iterator[List[str]]:
```

Comment:
this is much better now, but my idea had been to hook into the _insert function of the superclass. the problem you have now is, that your implementation does not respect self._sql_client.capabilities.max_query_length and self._sql_client.capabilities.max_rows_per_insert. that's why I suggested to brancht out at "if max_rows is not None:" and not at the beginning of this method.

Reply:
@sh-rp Thanks for feedback -- apologies for not implementing your idea correctly.  Is this better?  Still WIP ... am I branching out correctly?